### ISSUE TITLE
Allow Android arm32 apps to run on Android arm64 devices

### ISSUE THREAD

**Issue by fanyang-mono** [dotnet/xharness#584]
Currently, if I try to run Android arm32 test apps on Android arm64 devices, I ran into the following error
```
[08:57:43] fail: No devices with architecture 'armeabi-v7a' was found among attached devices.
      [08:57:43] crit: Failed to run test package: Failed to find compatible device: armeabi-v7a
                       System.Exception: Failed to run test package: Failed to find compatible device: armeabi-v7a
                          at Microsoft.DotNet.XHarness.CLI.Commands.Android.AndroidInstallCommand.InvokeHelper(ILogger logger, String apkPackageName, String appPackagePath, String apkRequiredArchitecture, String deviceId, TimeSpan bootTimeoutSeconds, AdbRunner runner) in /_/src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidInstallCommand.cs:line 120
                          at Microsoft.DotNet.XHarness.CLI.Commands.Android.AndroidTestCommand.InvokeInternal(ILogger logger) in /_/src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidTestCommand.cs:line 77
```
However, theoretically, it should be allowed, according to android docs:
>  64-bit devices also support their 32-bit variants. Using arm64-v8a devices as an example, the device can also run armeabi and armeabi-v7a code

**Comment by SamMonoRT** [comment]
This is blocking enabling test validation in CI on Arm32 apps.

**Comment by greenEkatherine** [comment]
Looking into it now

**Comment by greenEkatherine** [comment]
@fanyang-mono meanwhile as a workaround you may put explicit parameter `--device-arch="arm64-v8a"` to force running on these devices

**Comment by greenEkatherine** [comment]
@fanyang-mono @SamMonoRT could you please provide test APK for this?

**Comment by greenEkatherine** [comment]
This problem might require two changes. The first one - and I started with it - continue looking for compatible devices if we detect more then one architecture from APK. As a bonus of this change there will be possible to provide `--device-arch` more than once. I think it makes total sense because application may target more than one architecture.

Then I need APK to test if the first part would be enough. In other case the second part requires to check not only device architecture itself but its compatible architectures as well. In my opinion it should take effect only when `--device-arch` isn't set explicitly. Otherwise it might mean that user want exact that architecture and not some compatible one.

**Comment by greenEkatherine** [comment]
I've raised PR for the first part https://github.com/dotnet/xharness/pull/591

**Comment by fanyang-mono** [comment]
> @fanyang-mono @SamMonoRT could you please provide test APK for this?

Talked to Katya offline. She will use one of the apk that she has.

**Comment by greenEkatherine** [comment]
I've merged https://github.com/dotnet/xharness/pull/591 it should fix the issue. If not please let me know, I will prepare change to check compatibility from the side of a device as well.

**Comment by greenEkatherine** [comment]
XHarness version with fix `1.0.0-prerelease.21257.1` https://dev.azure.com/dnceng/public/_packaging?_a=package&feed=dotnet-eng&package=Microsoft.DotNet.XHarness.CLI&version=1.0.0-prerelease.21257.1&protocolType=NuGet&view=overview
